[Fleet] add `ingest-agent-policies.is_protected` SO attribute

joeypoon opened the pull request:
## Summary

add `ingest-agent-policies.is_protected` SO attribute


### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios


### For maintainers

- [x] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

nchaulet commented on x-pack/plugins/fleet/common/openapi/components/schemas/agent_policy_create_request.yaml:
```diff
@@ -44,6 +44,8 @@ properties:
       required:
         - name
         - enabled
+  is_protected:
```
Do you mind adding a description of what `is_protected` do here?

nchaulet reviewed (approved):
Looks like there is some tests to fix but otherwise 🚀 

nit: It will be great to add some description in the PR of what `is_protected` is supposed to do.

joeypoon commented:
@nchaulet I added some [license gating to the handlers](https://github.com/elastic/kibana/pull/157077/commits/dff873b8c195f5b07b460c16d54d7978fa0bfc33), would you mind taking another look?

I really wanted to add some tests for the new license gating scenarios but to my knowledge, we don't really have a way to switch licenses on the fly within our FTR tests? I tested all the scenarios locally. Alternatively, I also considered just stripping the `is_protected` argument if insufficient license level. Though we aren't exposing the `is_protected` toggle on the UI unless they have sufficient license so I figured it was low risk.

nchaulet commented:
> I really wanted to add some tests for the new license gating scenarios but to my knowledge, we don't really have a way to switch licenses on the fly within our FTR tests?

Yes we cannot really switch licence in FTR tests, what about adding the licence check to the agent policy service this way the check can easily be unit tested? Also a policy can programatically be created outside of the API through kibana config file so having that check here probably make sense.

paul-tavares reviewed (approved):
👍

joeypoon commented:
> Yes we cannot really switch licence in FTR tests, what about adding the licence check to the agent policy service this way the check can easily be unit tested? Also a policy can programatically be created outside of the API through kibana config file so having that check here probably make sense.

@nchaulet thanks for the insight, I hadn't considered policy creation through kibana config. I've moved the license gating to the agent policy service and added unit tests.

nchaulet reviewed (approved):
LGTM 🚀
